ui.apps/src/main/content/jcr_root/apps/etoolbox-authoring-kit/policies/js/maxChildren.js

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Utility that disables insertion, drag/drop and copy/paste actions for container component
+ * if there are more child components inserted than the {@link MaxChildrenLimiter.LIMIT_RESOLVER_NAME} listener returns or
+ * defined by {@link MaxChildrenLimiter.LIMIT_RESOLVER_PROPERTY} in policies or designs.
+ */
+(function (ns, $, author) {
+    'use strict';
+
+    ns.MaxChildrenLimiter = ns.MaxChildrenLimiter || {};
+
+    /** The name of listener to resolve parsys children limit */
+    ns.MaxChildrenLimiter.LIMIT_RESOLVER_NAME = 'resolvemaxchildern';
+
+    /** The name of property to resolve parsys children limit */
+    ns.MaxChildrenLimiter.LIMIT_RESOLVER_PROPERTY = 'eak-max-children';
+
+    /** The default limit active message */
+    ns.MaxChildrenLimiter.ERROR_MESSAGE = ns.I18n.get('Maximum number of allowed components reached');
+
+    /** The name of attribute to store limit message and marker */
+    ns.MaxChildrenLimiter.ERROR_MESSAGE_ATTR = 'data-max-children-marker';
+
+    /**
+     * @param {Editable} editable
+     * @returns {boolean} true if editable is a 'newpar' parsys zone
+     */
+    ns.MaxChildrenLimiter.isPlaceholder = (editable) => editable && editable.type && editable.type.endsWith('newpar');
+
+    /**
+     * @param editable
+     * @returns {number} - children limit for the given editable
+     */
+    ns.MaxChildrenLimiter.getChildrenLimit = function getChildrenLimit(editable) {
+        const limit = ns.EAKUtils.executeListener(editable, ns.MaxChildrenLimiter.LIMIT_RESOLVER_NAME);
+        if (typeof limit === 'number' || typeof limit === 'string') return +limit;
+        return ns.MaxChildrenLimiter.resolveChildrenLimitFromPolicy(editable);
+    };
+
+    /**
+     * @param editable
+     * @returns {number} - children limit for the given editable
+     */
+    ns.MaxChildrenLimiter.resolveChildrenLimitFromPolicy = function resolveChildrenLimitFromPolicy(editable) {
+        const limitCfg = ns.EAKPolicyUtils.findPropertyFromConfig(editable, ns.MaxChildrenLimiter.LIMIT_RESOLVER_PROPERTY);
+        return limitCfg === null ? Number.POSITIVE_INFINITY : +limitCfg;
+    };
+
+    /**
+     * @param editable
+     * @returns {number} current children count for the given editable
+     */
+    ns.MaxChildrenLimiter.getChildrenCount = function getChildrenCount(editable) {
+        if (!editable.dom) return 0;
+        const children = editable.dom.children(':not(cq, .par, .newpar, .iparys_inherited)');
+        return children ? children.length : 0;
```

Comment:
```suggestion
        return children.length || 0;
```

Reply:
Nope, that was, actually, a check for `children = null`, but it is useless as well ) 
I will take a look, thanks!